plugins/inputs/http_listener_v2/http_listener_v2.go

```diff
@@ -305,6 +307,14 @@ func (h *HTTPListenerV2) serveWrite(res http.ResponseWriter, req *http.Request)
 			m.AddTag(pathTag, req.URL.Path)
 		}
 
+		if h.QueryTag {
+			for key, values := range queryParams {
+				if len(values) > 0 {
+					m.AddTag(key, values[0])  // Only use the first value of the query parameter
```

Comment:
Are there cases where we would need more than the first value?

Reply:
Actually, I think it's not. It was a quick way to parse the value because it's an array, but now that I looked at it more carefully I consider that it could be possible to include another loop to parse it. Do you think this approach is more accurate?

```go
if h.QueryTag {
	queryParams := req.URL.Query()
	for key, values := range queryParams {
		if len(values) > 0 {
			for _, value := range values {
				m.AddTag(key, value)
			}
		}
	}
}
```

Reply:
I think it does make more sense to be fully accurate, though I worry about this being specifically a tag, and cardinality issues it causes. If the query values change with any regularity this could cause issues. Do you think it makes sense to have this as a field instead?

Reply:
> Maybe we could give the option to choose what parameters are tags, like you currently do with the option `http_header_tags`. Then another option to choose some parameters as fields. What do you think about?

I think this is a good solution, @srebhan what do you think?

Reply:
I think having `http_query_parameter_tags` might be good. This way the user has control over the cardinality. However, if multiple values are present, how about having one tag, sort the multiple values and string-join them?